File: MatchRiderGO/LocationBadge.m
//
//  LocationBadge.m
//  MatchRiderGO
//
//  Created by Leye on 29.02.16.
//  Copyright (c) 2016 LivelyCode. All rights reserved.
//

#import "LocationBadge.h"

@implementation LocationBadge

- (id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        self.edgeInsets = UIEdgeInsetsMake(5, 10, 5, 10);
    }
    return self;
}


// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
    [super drawTextInRect:UIEdgeInsetsInsetRect(rect, self.edgeInsets)];
}



@end

File: MatchRiderGO/LocationBadge.swift
//
//  LocationBadge.swift
//  MatchRiderGO
//
//  Created by Leye on 03.03.16.
//  Copyright (c) 2016 LivelyCode. All rights reserved.
//

import UIKit

class LocationBadge: UILabel {
    required init(coder aDecoder: NSCoder) {
        super.init(coder:aDecoder)
        self.setup()
    }
    
    override init(frame:CGRect) {
        super.init(frame:frame)
        self.setup()
    }
    
    override  func awakeFromNib() {
        super.awakeFromNib()
        self.setup()
    }
    override class func layerClass() -> AnyClass {
        return CATextLayer.self
    }
    
    func textLayer() -> CATextLayer {
        return self.layer as CATextLayer
    }
    
    func setup() {
        self.text = self.text
        self.layer.masksToBounds = true
        self.layer.cornerRadius = 5
        
        self.textLayer().alignmentMode = kCAAlignmentJustified
        self.textLayer().wrapped = true
        self.layer.display()
    }
    
}
